app/src/main/java/org/oppia/app/player/state/StateFragment.kt

```diff
@@ -5,12 +5,25 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.databinding.ObservableField
 import org.oppia.app.fragment.InjectableFragment
+import org.oppia.app.player.audio.CellularDataDialogFragment
+import org.oppia.app.player.audio.CellularDataInterface
 import javax.inject.Inject
 
-/** Fragment that represents the current state card of an exploration. */
-class StateFragment : InjectableFragment() {
-  @Inject lateinit var stateFragmentPresenter: StateFragmentPresenter
+private const val TAG_CELLULAR_DATA_DIALOG = "CELLULAR_DATA_DIALOG"
+
+/** Fragment that represents the current state of an exploration. */
+class StateFragment : InjectableFragment(), CellularDataInterface {
+  @Inject
+  lateinit var stateFragmentPresenter: StateFragmentPresenter
+  // Control this boolean value from controllers in domain module.
+  private var showCellularDataDialog = true
+  var isAudioFragmentShowing = ObservableField<Boolean>(false)
```

Comment:
Can you add tests to verify from the perspective of StateFragment that driving through the two dialog flows properly shows/hides the audio fragment in the expected situations?

Reply:
I think my original comment wasn't very clear. I was actually looking for the two cellular dialog cases where pressing okay/cancel plays/doesn't play audio, but I realize we probably can't test that yet.

However, this test doesn't seem to test that the cellular dialog actually opens. Can you add tests to verify that the cellular dialog can open, and that pressing okay/cancel (2 tests) properly dismiss the dialog?

Reply:
Added test cases for two scenarios:

1. CellularDialogShows -> Click Ok
2. CellularDialogShows -> Click Cancel